skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,322 @@
+ */
+package slack.tooling.projectgen.circuitgen
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import com.squareup.kotlinpoet.*
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import java.io.File
+import java.io.StringWriter
+
+interface CircuitComponent {
+  val fileSuffix: String
+
+  fun generate(packageName: String, className: String): FileSpec
+
+  fun isTestComponent(): Boolean = false
+
+  fun writeToFile(directory: String, packageName: String?, className: String) {
+    val baseDirectory =
+      if (isTestComponent()) directory.replace("src/main", "src/test") else directory
+    val fileSpec = generate(packageName.orEmpty(), className)
+    File(baseDirectory).apply { if (!exists()) mkdirs() }
+    val stringWriter = StringWriter()
+    fileSpec.writeTo(stringWriter)
+    val generatedCode = stringWriter.toString().replace("public ", "")
+    File("${baseDirectory}/${className}${fileSuffix}.kt").writeText(generatedCode)
+  }
+}
+
+class CircuitScreen : CircuitComponent {
+  override val fileSuffix = "Screen"
+
+  override fun generate(packageName: String, className: String): FileSpec {
+    /*
+    Generate Circuit Screen class, eg:
+
+    @Parcelize
+      class FeatureScreen : Screen {
+        data class State(
+          val message: String = "",
+          val eventSink: Event.() -> Unit = {},
+        ) : CircuitUiState
+
+        @Immutable
+        sealed interface Event : CircuitUiEvent
+      }
+    */
+    val circuitScreenClass = "${className}Screen"
+    val parcelize = ClassName("kotlinx.parcelize", "Parcelize")
+    val circuitUiEvent = ClassName("com.slack.circuit.runtime", "CircuitUiEvent")
+    val circuitUiState = ClassName("com.slack.circuit.runtime", "CircuitUiState")
+    val screenInterface = ClassName("com.slack.circuit.runtime.screen", "Screen")
+    val eventInterface =
+      TypeSpec.interfaceBuilder("Event")
+        .addAnnotation(Immutable::class)
+        .addModifiers(KModifier.SEALED)
+        .addSuperinterface(circuitUiEvent)
+        .build()
+
+    val eventInterfaceName = eventInterface.name?.let { ClassName("", it) }
+    val stateClass =
+      TypeSpec.classBuilder("State")
+        .addModifiers(KModifier.DATA)
+        .primaryConstructor(
+          FunSpec.constructorBuilder()
+            .addParameter(
+              ParameterSpec.builder("message", String::class).defaultValue("%S", "").build()
+            )
+            .addParameter(
+              ParameterSpec.builder(
+                  "eventSink",
+                  LambdaTypeName.get(
+                    receiver = eventInterfaceName,
+                    returnType = Unit::class.asTypeName(),
+                  ),
+                )
+                .defaultValue("{}")
+                .build()
+            )
+            .build()
+        )
+        .addProperty(PropertySpec.builder("message", String::class).initializer("message").build())
```

Comment:
Let's not include this property, for the same reason as other dummy data - people will check it in :)